pkg/engine/replica/diff_disk.go

```diff
@@ -207,11 +207,25 @@ func (d *diffDisk) fullReadAt(buf []byte, offset int64) (int, error) {
 	return count, nil
 }
 
-func (d *diffDisk) read(target byte, buf []byte, offset int64, startSector int64, sectors int64) (int, error) {
+func (d *diffDisk) read(target byte, buf []byte, startOffset int64, startSector int64, sectors int64) (int, error) {
```

Comment:
Test: Launch a longhorn block device and read the out-of-bound part to see if it will error out